pkg/kv/kvserver/replica_proposal.go

```diff
@@ -263,9 +263,12 @@ func (r *Replica) leasePostApplyLocked(
 	leaseChangingHands := prevLease.Replica.StoreID != newLease.Replica.StoreID || prevLease.Sequence != newLease.Sequence
 
 	if iAmTheLeaseHolder {
-		// Log lease acquisition whenever an Epoch-based lease changes hands (or verbose
-		// logging is enabled).
-		if newLease.Type() == roachpb.LeaseEpoch && leaseChangingHands || log.V(1) {
+		// Log lease acquisition whenever:
+		// 1. epoch-based lease changes hands
+		// 2. the new leaseholder is draining
+		// 3. verbose logging is enabled
+		epochLeaseChangingHands := newLease.Type() == roachpb.LeaseEpoch && leaseChangingHands
+		if epochLeaseChangingHands || r.store.IsDraining() || log.V(1) {
 			log.VEventf(ctx, 1, "new range lease %s following %s", newLease, prevLease)
```

Comment:
Should this be a log.Infof instead?

Reply:
Good catch. This logic was slowly adapted by https://github.com/cockroachdb/cockroach/commit/c3dd4724de85726fd4fac4ba02828f48622295e6 and https://github.com/cockroachdb/cockroach/commit/c3dd4724de85726fd4fac4ba02828f48622295e6. I think the change I made strikes the desired balance of verbosity, but PTAL.